### PR TITLE
Require for loop init rewriter for SSA.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Review by bshastry — DISMISSED** [review]
LGTM

File: libyul/optimiser/SSATransform.h [hunk 1]
```diff
@@ -85,7 +85,7 @@ class NameDispenser;
  *
  * TODO Which transforms are required to keep this idempotent?
  *
- * Prerequisite: Disambiguator.
+ * Prerequisite: Disambiguator, For loop init rewriter.
```
Comment (hrkrshnn) [inline libyul/optimiser/SSATransform.h]:
```suggestion
 * Prerequisite: Disambiguator, ForLoopInitRewriter.
```

For consistency, with the rest of the files.

**Review by ekpyron — DISMISSED** [review]
Good apart from @hrkrshnn's comment https://github.com/ethereum/solidity/pull/9295/files#r449448638!

**Comment by ekpyron** [comment]
I just commited and squashed @hrkrshnn comment myself.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**